### PR TITLE
Getting better Locators

### PR DESCRIPTION
**Pull request by kliver98**
Local execution:
image
Headless execution:
image

### CODE REVIEW

File: src/page/address-step.page.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export class AddressStepPage {
   private proceedToCheckout: ElementFinder;
 
   constructor () {
-    this.proceedToCheckout = $('#center_column > form > p > button > span');
+    this.proceedToCheckout = $('form .cart_navigation > button[type="submit"]');
```
Comment (holgiosalos) [inline src/page/address-step.page.ts]: could you try to use a different attribute than type? for example, name attribute, will not be susceptible to changes and give us more context about what we are looking for

File: src/page/bank-payment.page.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ export class BankPaymentPage {
   private confirmButton: ElementFinder;
 
   constructor () {
-    this.confirmButton = $('#cart_navigation > button > span');
+    this.confirmButton = $('#cart_navigation > button[type="submit"]');
```
Comment (holgiosalos) [inline src/page/bank-payment.page.ts]: could you try to use a different attribute than type? for example, name attribute, will not be susceptible to changes and give us more context about what we are looking for

File: src/page/menu-content.page.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ export class MenuContentPage {
   private tShirtMenu: ElementFinder;
 
   constructor () {
-    this.tShirtMenu = $('#block_top_menu > ul > li:nth-child(3) > a');
+    this.tShirtMenu = $('#block_top_menu > ul > li:nth-child(3)');
```
Comment (holgiosalos) [inline src/page/menu-content.page.ts]: Use a different selector, this is depending on the position of the T-Shirts option, you can try to use the title attribute and also some of the existing elements in the current locator.

File: src/page/product-list.page.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ export class ProductListPage {
   private addToCart: ElementFinder;
 
   constructor () {
-    this.addToCart = $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default');
+    this.addToCart = $('ul.product_list > li:nth-child(1) .button-container .ajax_add_to_cart_button');
```
Comment (holgiosalos) [inline src/page/product-list.page.ts]: avoid using the selectors of type nth-child, they depend on the structure of the HTML, which can make the tests flaky, try to find an attribute or property that indicates we are clicking the add to cart button

File: src/page/shipping-step.page.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ export class ShippingStepPage {
 
   constructor () {
     this.termsOfService = $('#cgv');
-    this.proceedToCheckout = $('#form > p > button > span');
+    this.proceedToCheckout = $('form .cart_navigation > button[type="submit"]');
```
Comment (holgiosalos) [inline src/page/shipping-step.page.ts]: could you try to use a different attribute than type? for example, name attribute, will not be susceptible to changes and give us more context about what we are looking for

File: src/page/sign-in-step.page.ts [hunk 6]
```diff
@@ -6,9 +6,9 @@ export class SignInStepPage {
   private loginSubmitButton: ElementFinder;
 
   constructor () {
-    this.emailField = $('#email');
-    this.passwdField = $('#passwd');
-    this.loginSubmitButton = $('#SubmitLogin > span');
+    this.emailField = $('#login_form #email');
+    this.passwdField = $('#login_form #passwd');
+    this.loginSubmitButton = $('#login_form #SubmitLogin');
```
Comment (holgiosalos) [inline src/page/sign-in-step.page.ts]: we don't need to change the email and password locators. And about the submit button, you can simplify it :)

File: src/page/summary-step.page.ts [hunk 7]
```diff
@@ -4,7 +4,7 @@ export class SummaryStepPage {
   private proceedToCheckout: ElementFinder;
 
   constructor () {
-    this.proceedToCheckout = $('.cart_navigation span');
+    this.proceedToCheckout = $('.cart_navigation .standard-checkout');
```
Comment (holgiosalos) [inline src/page/summary-step.page.ts]: try to find a locator that uses the title attribute

**Review by holgiosalos — CHANGES_REQUESTED** [review]
You can also improve the locators for:
- src/page/order-summary.page.ts
- src/page/product-added-modal.page.ts

File: src/page/summary-step.page.ts (hunk 7 above)
Comment (kliver98) [reply src/page/summary-step.page.ts]: Done

File: src/page/sign-in-step.page.ts (hunk 6 above)
Comment (kliver98) [reply src/page/sign-in-step.page.ts]: Done

File: src/page/shipping-step.page.ts (hunk 5 above)
Comment (kliver98) [reply src/page/shipping-step.page.ts]: Done

File: src/page/product-list.page.ts (hunk 4 above)
Comment (kliver98) [reply src/page/product-list.page.ts]: Done

File: src/page/menu-content.page.ts (hunk 3 above)
Comment (kliver98) [reply src/page/menu-content.page.ts]: Done

File: src/page/bank-payment.page.ts (hunk 2 above)
Comment (kliver98) [reply src/page/bank-payment.page.ts]: Done

File: src/page/address-step.page.ts (hunk 1 above)
Comment (kliver98) [reply src/page/address-step.page.ts]: Done

File: src/page/product-list.page.ts [hunk 8]
```diff
@@ -4,7 +4,7 @@ export class ProductListPage {
   private addToCart: ElementFinder;
 
   constructor () {
-    this.addToCart = $('ul.product_list > li:nth-child(1) .button-container .ajax_add_to_cart_button');
+    this.addToCart = $('.button-container [data-id-product="1"]');
```
Comment (holgiosalos) [inline src/page/product-list.page.ts]: what about `['#center_column [title="Add to cart"]'`

**Review by holgiosalos — APPROVED** [review]

File: src/page/product-list.page.ts (hunk 8 above)
Comment (kliver98) [reply src/page/product-list.page.ts]: Better. Done

**Pull request merged**